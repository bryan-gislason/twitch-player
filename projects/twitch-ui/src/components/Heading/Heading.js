import PropTypes from "prop-types";
import React from "react";
import { StyleSheet } from "react-native";
import { useTheme } from "../../hooks";
import Text from "../Text";

const Heading = ({ level, children, style = [], testID = "heading" }) => {
  const { colors, layout, typography } = useTheme();

  const rootStyle = [
    styles.root,
    {
      color: colors.heading,
      fontSize:
        level === "1"
          ? typography.sizePrimary
          : level === "2"
          ? typography.sizeSecondary
          : typography.sizeTertiary,
      marginHorizontal: layout.gapExtraLarge,
      marginTop: layout.gapLarge,
      marginBottom: layout.gapSmall,
    },
  ].concat(style);

  return (
    <Text
      accessibilityRole="heading"
      aria-level={level}
      style={rootStyle}
      testID={testID}
    >
      {children}
    </Text>
  );
};

Heading.propTypes = {
  level: PropTypes.oneOf(["1", "2", "3"]).isRequired,
  children: PropTypes.node.isRequired,
  style: PropTypes.any,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    fontWeight: "bold",
  },
});

export default Heading;
