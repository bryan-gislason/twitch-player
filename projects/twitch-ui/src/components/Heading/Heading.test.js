import React from "react";
import { render } from "../../testing";
import Heading from "./Heading";

describe("Heading", () => {
  let instance;

  describe("level 1", () => {
    beforeEach(() => {
      instance = render(<Heading level="1">Some Heading</Heading>);
    });

    it("renders a level-1 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });

  describe("level 2", () => {
    beforeEach(() => {
      instance = render(<Heading level="2">Some Heading</Heading>);
    });

    it("renders a level-2 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });

  describe("level 3", () => {
    beforeEach(() => {
      instance = render(<Heading level="3">Some Heading</Heading>);
    });

    it("renders a level-3 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });
});
