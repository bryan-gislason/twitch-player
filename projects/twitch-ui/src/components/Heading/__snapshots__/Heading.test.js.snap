// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading level 1 renders a level-1 heading 1`] = `
<h1
  aria-level="1"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-a8ghvy rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 32px; margin: 20px 30px 5px 30px;"
>
  Some Heading
</h1>
`;

exports[`Heading level 2 renders a level-2 heading 1`] = `
<h2
  aria-level="2"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-a8ghvy rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 16px; margin: 20px 30px 5px 30px;"
>
  Some Heading
</h2>
`;

exports[`Heading level 3 renders a level-3 heading 1`] = `
<h3
  aria-level="3"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-a8ghvy rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 13px; margin: 20px 30px 5px 30px;"
>
  Some Heading
</h3>
`;
